planner/core/rule_derive_topn_from_window.go

```diff
@@ -38,14 +39,38 @@ func appendDerivedTopNTrace(topN LogicalPlan, opt *logicalOptimizeOp) {
 	opt.appendStepToCurrent(topN.ID(), topN.TP(), reason, action)
 }
 
+// checkPartitionBy mainly checks if partition by of window function is a prefix of
+// data order (clustered index) of the data source. TiFlash is allowed only for empty partition by.
+func checkPartitionBy(p *LogicalWindow, d *DataSource) bool {
+	// No window partition by. We are OK.
+	if len(p.PartitionBy) == 0 {
+		return true
+	}
+
+	// Table not clustered and window has partition by. Can not do the TopN piush down.
```

Comment:
```suggestion
	// Table not clustered and window has partition by. Can not do the TopN push down.
```
A typo here.

Reply:
Got it and will be fixed in next update.